indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -578,6 +578,10 @@ public void run()
               log.debug("All partitions have been fully read.");
               publishOnStop.set(true);
               stopRequested.set(true);
+
+              // We let the fireDepartmentMetrics know that all messages have been read. This way, some metrics such as
+              // high message gap need not be reported
+              fireDepartmentMetrics.markProcessingDone();
```

Comment:
should this be invoked in some other places as well?

Reply:
Oh hmm, since the task can publish segments in the middle of indexing, you need to check if the runner status is `PUBLISHING` (or ingestionState is `COMPLETED`) as well in the handoff notifier before counting down the latch. This is getting ugly, but I'm not sure if there is a cleaner way. We could implement a status change listener, but unsure if this is going to be useful in production.